src/c_api/c_api.cc

```diff
@@ -1313,10 +1313,8 @@ XGB_DLL int XGBoosterLoadModel(BoosterHandle handle, const char* fname) {
 
 namespace {
 void WarnOldModel() {
-  if (XGBOOST_VER_MAJOR >= 2) {
```

Comment:
The version is lowered since we have released an extra 1.7 release. The original plan was to go from 1.6 to 2.0